tests/generators/test_cycle.py

```diff
@@ -0,0 +1,55 @@
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+import unittest
+
+import retworkx
+
+
+class TestCycleGraph(unittest.TestCase):
+
+    def test_directed_cycle_graph(self):
+        graph = retworkx.generators.directed_cycle_graph(20)
+        self.assertEqual(len(graph), 20)
+        self.assertEqual(len(graph.edges()), 20)
+        for i in range(19):
+            self.assertEqual(graph.out_edges(i), [(i, i + 1, None)])
+        self.assertEqual(graph.out_edges(19), [(19, 0, None)])
+
+    def test_directed_cycle_graph_weights(self):
+        graph = retworkx.generators.directed_cycle_graph(
+            weights=list(range(20)))
```

Comment:
Weights are attached to nodes, rather than edges? And it's also separate from node data (which seems to be `None`, here)?

Reply:
Yeah, the weights here are attached to the nodes, the edge weights end up as `None` in the generator. I use data and weights interchangeably in retworkx (which is probably pretty sloppy on my part). The `None`s below are actually the edge weight/data `PyDiGraph.out_edges()` returns a list of tuples of the form `(source node index, target node index, edge weight)` https://github.com/Qiskit/retworkx/blob/master/src/digraph.rs#L852-L871. The `assertEqual` at L32 is checking the node weights, `PyDigraph.nodes()` returns a list of all the node data https://github.com/Qiskit/retworkx/blob/master/src/digraph.rs#L391-L401

Reply:
thank you:)